### PR TITLE
REGR: fix groupby std() with nullable dtypes

### LINKED ISSUE #37415: BUG: groupby with std aggregation of pandas integer dtype throws exception: 'IntegerArray' object has no attribute 'reshape'
- [x ] I have checked that this issue has not already been reported.

- [x ] I have confirmed this bug exists on the latest version of pandas.

- [ ] (optional) I have confirmed this bug exists on the master branch of pandas.

---

#### Code Sample, a copy-pastable example

```python
import pandas as pd
df = pd.DataFrame({'a': [1,1,1,2,2],'b': [1,2,3,4,5]})
df['b'] = df['b'].astype(pd.Int32Dtype())
df.groupby('a').agg('std') # Error 'IntegerArray' object has no attribute 'reshape'. The same error occurs with all pandas integer types

```

#### Problem description

For all integer pandas dtypes the code expects an IntegerArray to have the reshape function as is the case with numpy arrays.
It is possible that the error will also occur with other operations performed after a groupby, this has only been checked for 'std'.
The error occurs at
pandas\core\groupby\groupby.py", line 2491, in _get_cythonized_result
    vals = vals.reshape((-1, 1))

#### Expected Output

The same output as with a numpy data type, for example above this would be:

 
a 	 b
1 	1.000000
2 	0.707107

#### Output of ``pd.show_versions()``

<details>

INSTALLED VERSIONS
------------------
commit           : db08276bc116c438d3fdee492026f8223584c477
python           : 3.6.9.final.0
python-bits      : 64
OS               : Linux
OS-release       : 4.19.112+
Version          : #1 SMP Thu Jul 23 08:00:38 PDT 2020
machine          : x86_64
processor        : x86_64
byteorder        : little
LC_ALL           : None
LANG             : en_US.UTF-8
LOCALE           : en_US.UTF-8

pandas           : 1.1.3
numpy            : 1.18.5
pytz             : 2018.9
dateutil         : 2.8.1
pip              : 19.3.1
setuptools       : 50.3.0
Cython           : 0.29.21
pytest           : 3.6.4
hypothesis       : None
sphinx           : 1.8.5
blosc            : None
feather          : 0.4.1
xlsxwriter       : None
lxml.etree       : 4.2.6
html5lib         : 1.0.1
pymysql          : None
psycopg2         : 2.7.6.1 (dt dec pq3 ext lo64)
jinja2           : 2.11.2
IPython          : 5.5.0
pandas_datareader: 0.9.0
bs4              : 4.6.3
bottleneck       : 1.3.2
fsspec           : None
fastparquet      : None
gcsfs            : None
matplotlib       : 3.2.2
numexpr          : 2.7.1
odfpy            : None
openpyxl         : 2.5.9
pandas_gbq       : 0.13.2
pyarrow          : 0.14.1
pytables         : None
pyxlsb           : None
s3fs             : None
scipy            : 1.4.1
sqlalchemy       : 1.3.19
tables           : 3.4.4
tabulate         : 0.8.7
xarray           : 0.15.1
xlrd             : 1.1.0
xlwt             : 1.3.0
numba            : 0.48.0

</details>

### PR DESCRIPTION
**Pull request by jorisvandenbossche**
Fixes #37415

### CODE REVIEW

**Comment by jorisvandenbossche** [comment]
At some point we need to properly support nullable dtypes in the cython groupby operations (by passing+using the mask), but for now this ensure it at least works (although performance will not be ideal)

File: pandas/tests/groupby/aggregate/test_cython.py [hunk 1]
```diff
@@ -277,3 +277,38 @@ def test_read_only_buffer_source_agg(agg):
     expected = df.copy().groupby(["species"]).agg({"sepal_length": agg})
 
     tm.assert_equal(result, expected)
+
+
+@pytest.mark.parametrize(
```
Comment (simonjayhawkins) [inline pandas/tests/groupby/aggregate/test_cython.py:282]: could use all_numeric_reductions fixture (although slightly different) doesn't include `sem` but does include `kurt` and `skew`

Comment (jorisvandenbossche) [reply pandas/tests/groupby/aggregate/test_cython.py:282]: `kurt` is not implemented for groupby, it seems. I mainly copied this from another test in this file.

Comment (simonjayhawkins) [reply pandas/tests/groupby/aggregate/test_cython.py:282]: ok for now I guess.

**Review by simonjayhawkins — APPROVED** [review]
Thanks @jorisvandenbossche lgtm pending green.

(Did try to bisect to be sure of changes that caused regression but 1.0.0.rc raised `TypeError: cannot safely cast non-equivalent float64 to int32`)

**Comment by simonjayhawkins** [comment]
> (Did try to bisect to be sure of changes that caused regression but 1.0.0.rc raised `TypeError: cannot safely cast non-equivalent float64 to int32`)

rerun bisect with updated code sample for runner see https://github.com/pandas-dev/pandas/issues/37415#issuecomment-717182856, so fix seems reasonable

**Comment by jorisvandenbossche** [comment]
Ah, yes, sorry I should have mentioned, but I already narrowed it down to that PR. Thanks anyway!

Ci failure is unrelated I think

File: pandas/tests/groupby/aggregate/test_cython.py (hunk 1 above)
Comment (jreback) [reply pandas/tests/groupby/aggregate/test_cython.py:282]: yeah can you create an issue for folks to replace these with fixtures (good first)

File: pandas/tests/groupby/aggregate/test_cython.py [hunk 2]
```diff
@@ -277,3 +277,38 @@ def test_read_only_buffer_source_agg(agg):
...
+        {
+            "A": ["A", "B"] * 5,
+            "B": pd.array([1, 2, 3, 4, 5, 6, 7, 8, 9, pd.NA], dtype="Int64"),
+        }
+    )
+    result = getattr(df.groupby("A")["B"], op_name)()
+    df2 = df.assign(B=df["B"].astype("float64"))
+    expected = getattr(df2.groupby("A")["B"], op_name)()
+
+    if op_name != "count":
```
Comment (jreback) [inline pandas/tests/groupby/aggregate/test_cython.py:310]: yeah also an issue to fix this up (I think we have another issue about this) as this is slightly tricky, e.g. are we returning a nullable for the count itself (I don't know if we ever decided that)

**Comment by jreback** [comment]
thanks @jorisvandenbossche

**Comment by simonjayhawkins** [comment]
@meeseeksdev backport 1.1.x

**Comment by jorisvandenbossche** [comment]
For follow-ups, I opened https://github.com/pandas-dev/pandas/issues/37493 for general "native" support for masked arrays in groupby algos, and https://github.com/pandas-dev/pandas/issues/37494 for the specific issue to use the correct dtype for the result.

**Pull request merged**